packages/salesforcedx-vscode-apex/package.json

```diff
@@ -268,6 +268,11 @@
           "type": "boolean",
           "default": false,
           "description": "%enable_sobject_refresh_on_startup_description%"
+        },
+        "salesforcedx-vscode-apex.advanced.enable-completion-statistics": {
```

Comment:
Bryan mentioned this already, if you don't want this to be surfaced in the UI just remove it completely from the `package.json`. That will make the extension rely on code to know how to handle the setting but we've done it before, take a look at `internal-development` in the core extension.